### PR TITLE
Update workbench documentation links, complete renaming

### PR DESCRIPTION
**Pull request by davidcui-amzn**
Signed-off-by: David Cui <davidcui@amazon.com>

### Description
* Update documentation links to OpenSearch
* Rename remaining occurrences of `opendistro`
 
### Issues Resolved
[List any issues this PR will resolve]
 
### Check List
- [ ] New functionality includes testing.
  - [x] All tests pass, including unit test, integration test and doctest
- [ ] New functionality has been documented.
  - [ ] New functionality has javadoc added
  - [ ] New functionality has user manual doc added
- [x] Commits are signed per the DCO using --signoff 

By submitting this pull request, I confirm that my contribution is made under the terms of the Apache 2.0 license.
For more information on following Developer Certificate of Origin and signing off your commits, please check [here](https://github.com/opensearch-project/OpenSearch/blob/main/CONTRIBUTING.md#developer-certificate-of-origin).

### CODE REVIEW

File: workbench/common/index.ts [hunk 1]
```diff
@@ -24,5 +24,5 @@
  *   permissions and limitations under the License.
  */
 
-export const PLUGIN_ID = 'opendistroQueryWorkbench';
+export const PLUGIN_ID = 'openSearchQueryWorkbench';
```
Comment (joshuali925) [inline workbench/common/index.ts:27]: this is not used in the plugin, but should be `queryWorkbenchDashboards`

File: workbench/server/plugin.ts [hunk 2]
```diff
@@ -46,7 +46,7 @@ export class WorkbenchPlugin implements Plugin<WorkbenchPluginSetup, WorkbenchPl
   }
 
   public setup(core: CoreSetup) {
-    this.logger.debug('opendistroQueryWorkbench: Setup');
+    this.logger.debug('openSearchQueryWorkbench: Setup');
```
Comment (joshuali925) [inline workbench/server/plugin.ts:49]: same here

File: workbench/server/plugin.ts [hunk 3]
```diff
@@ -62,7 +62,7 @@ export class WorkbenchPlugin implements Plugin<WorkbenchPluginSetup, WorkbenchPl
   }
 
   public start(core: CoreStart) {
-    this.logger.debug('opendistroQueryWorkbench: Started');
+    this.logger.debug('openSearchQueryWorkbench: Started');
```
Comment (joshuali925) [inline workbench/server/plugin.ts:65]: same here

**Review by joshuali925 — COMMENTED** [review]
will there be a change on renaming `_opensearch` to `_plugins` in API? https://github.com/opensearch-project/sql/blob/2ea5df6bf7a5e5abcd3f990a7691330adb6a7e11/workbench/server/services/utils/constants.ts#L29-L32

**Comment by davidcui-amzn** [comment]
> will there be a change on renaming `_opensearch` to `_plugins` in API?
> 
> https://github.com/opensearch-project/sql/blob/2ea5df6bf7a5e5abcd3f990a7691330adb6a7e11/workbench/server/services/utils/constants.ts#L29-L32

Good catch, I missed that. I assume you're referring to https://github.com/opensearch-project/opensearch-plugins/blob/main/UPGRADING.md#rest-apis-backward-compatibility ?

**Comment by dai-chen** [comment]
> > will there be a change on renaming `_opensearch` to `_plugins` in API?
> > https://github.com/opensearch-project/sql/blob/2ea5df6bf7a5e5abcd3f990a7691330adb6a7e11/workbench/server/services/utils/constants.ts#L29-L32
> 
> Good catch, I missed that. I assume you're referring to https://github.com/opensearch-project/opensearch-plugins/blob/main/UPGRADING.md#rest-apis-backward-compatibility ?

I've posted PR for that: https://github.com/opensearch-project/sql/pull/66

**Review by joshuali925 — APPROVED** [review]

**Review by chloe-zh — APPROVED** [review]

**Pull request merged**